lib/HLSL/LiveValueAnalysis.cpp

```diff
@@ -0,0 +1,1323 @@
+                      Phis.pop_back();
+                    }
+                  }
+                  TmpStr << "Use Locations:\n";
+                  PrettyStr << TmpStr.str();
+                  VSStr << TmpStr.str();
+
+                  // Find max path length to format horizontal dividers in the output.
+                  unsigned int maxPathLength = 0;
+                  unsigned int maxFuncLength = 0;
+                  for( auto Loc : UseLocations ) {
+                    std::string Path =
+                      formatSourceLocation( m_rootPath, Loc->getFilename().str(),
+                        Loc->getLine() );
+                    if( Path.length() > maxPathLength )
+                      maxPathLength = Path.length();
+                    if( Loc->getScope()
+                      ->getSubprogram()
+                      ->getName()
+                      .str()
+                      .length() > maxFuncLength )
+                      maxFuncLength = Loc->getScope()
+                      ->getSubprogram()
+                      ->getName()
+                      .str()
+                      .length();
+                    if( DILocation *InLoc = Loc->getInlinedAt() ) {
+                      std::string Path = formatSourceLocation(
+                        m_rootPath, InLoc->getFilename().str(),
+                        InLoc->getLine() );
+                      if( Path.length() > maxPathLength )
+                        maxPathLength = Path.length();
+                      if( InLoc->getScope()
+                        ->getSubprogram()
+                        ->getName()
+                        .str()
+                        .length() > maxFuncLength )
+                        maxFuncLength = InLoc->getScope()
+                        ->getSubprogram()
+                        ->getName()
+                        .str()
+                        .length();
+                      while( DILocation *NestedInLoc = InLoc->getInlinedAt() ) {
+                        InLoc = NestedInLoc;
+                        std::string Path = formatSourceLocation(
+                          m_rootPath, InLoc->getFilename().str(),
+                          InLoc->getLine() );
+                        if( Path.length() > maxPathLength )
+                          maxPathLength = Path.length();
+                        if( InLoc->getScope()
+                          ->getSubprogram()
+                          ->getName()
+                          .str()
+                          .length() > maxFuncLength )
+                          maxFuncLength = InLoc->getScope()
+                          ->getSubprogram()
+                          ->getName()
+                          .str()
+                          .length();
+                      }
+                    }
+                  }
+                  // Pad with spaces for inline text, etc
+                  maxFuncLength += 2;
+                  maxPathLength += 19;
```

Comment:
what is 19?

Reply:
A magic number representing the padding needed for the pretty print formatting below.  I can remove and add a const char* for the " inlined ..." and expound on the comment.